dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/webmodules/WebModule.java

```diff
@@ -74,17 +73,23 @@ public WebModule(String name, String namespace, String defaultAction) {
   }
 
   public static WebModule getModule(App app) {
+    String basePath = app.getBasePath();
+    if (app.isBundled()) {
+      basePath = "/api/apps/" + app.getKey();
+    }
+
     boolean hasIcon = app.getIcons() != null && app.getIcons().getIcon48() != null;
 
     String defaultAction = app.getLaunchUrl();
 
-    String icon = hasIcon ? app.getBaseUrl() + "/" + app.getIcons().getIcon48() : null;
+    String icon =
```

Comment:
It looks like the icon problems stems from here, since we go from using getBaseUrl, to basePath; BasePath is a relative url, while baseUrl is a complete url with the base url of the instance included.